libtcmu_config.c

```diff
@@ -180,6 +180,7 @@ static void tcmu_conf_free_str_keys(struct tcmu_config *cfg)
 	 * For example:
 	 * TCMU_FREE_CFG_STR_KEY(cfg, 'STR KEY');
 	 */
+	 TCMU_FREE_CFG_STR_KEY(cfg, log_dir_path);
```

Comment:
I think if you do the above then you need to fix tcmu_destroy_config:

```
        tcmu_conf_free_str_keys(cfg);
        if (cfg->log_dir_path)
                free(cfg->log_dir_path);
```
because tcmu_conf_free_str_keys call will delete cfg->log-dir_path and then the call above will try to delete the same pointer.

I actually thought the leak was if tcmu_conf_set_options is called twice. The first call would do:

`                cfg->key = strdup(option->opt_str); \`

but then the second call would see the cfg->key is set and we would end up allocating memory again without freeing and leak the old memory at cfg->key. I thought we needed a 

```
if (!cfg->key)
     cfg->key = strdup(option->opt_str); 
```